src/main/java/redis/clients/jedis/JedisPubSub.java

```diff
@@ -146,10 +150,18 @@ private void process(Client client) {
         onUnsubscribe(strchannel, subscribedChannels);
       } else if (Arrays.equals(MESSAGE.getRaw(), resp)) {
         final byte[] bchannel = (byte[]) reply.get(1);
-        final byte[] bmesg = (byte[]) reply.get(2);
         final String strchannel = (bchannel == null) ? null : SafeEncoder.encode(bchannel);
-        final String strmesg = (bmesg == null) ? null : SafeEncoder.encode(bmesg);
-        onMessage(strchannel, strmesg);
+        Object bmesg = reply.get(2);
+        //THe server sends a List rather than a byte[] on the redis invalidate channel;
+        if(bmesg instanceof List) {
+          final Object strmesg = (bmesg == null) ? null : SafeEncoder.encodeObject(bmesg);
+          List <String> strmesgn = (List <String> ) strmesg;
```

Comment:
This can fail. The channel can send a message of following type : ["a", "b", ["c", "d"]] as per protocol. If the user then iterates the list assuming it to be of String, it will result in a class cast exception.

We can assume and cast it to a list of string for one instance by using the fact that invalidation messages are list of keys